dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -124,14 +126,41 @@ namespace winrt::Microsoft::Windows::ToastNotifications::implementation
         THROW_HR_IF(E_UNEXPECTED, notificationId == 0);
         toast.ToastId(notificationId);
     }
-    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData /* data */, hstring /* tag */, hstring /* group */)
+
+    // API updates notification data when both tag and group are provided or when tag is provided.
+    // For other scenarios the API throws
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData const& data, hstring const& tag, hstring const& group)
     {
-        throw hresult_not_implemented();
+        if ((tag == winrt::hstring(L"") && group == winrt::hstring(L"")) || (tag == winrt::hstring(L"") && group != winrt::hstring(L"")))
```

Comment:
We return `NotificationNotFound` if:

`tag == winrt::hstring(L"") && group == winrt::hstring(L"")`
OR
`tag == winrt::hstring(L"") && group != winrt::hstring(L"")`

which means you can get rid of the group check, it doesn't matter.

But, isn't a valid a combination an empty tag but valid group?